std/hxcoro/AbstractTask.hx

```diff
@@ -21,21 +62,36 @@ class TaskException extends Exception {}
 	and should be kept in a state where it could even be moved outside the hxcoro package. Also, `state` should
 	be treated like a truly private variable and only be modified from within this class.
 **/
-abstract class AbstractTask<T> {
+abstract class AbstractTask<T> implements ICancellationToken {
 	final children:Array<AbstractTask<Any>>;
+	final cancellationCallbacks:Array<CancellationHandle>;
+	final noOpCancellationHandle:NoOpCancellationHandle;
```

Comment:
This looks like it should become a static.

Reply:
I've moved it into a static.